src/stream/cycle.rs

```diff
@@ -0,0 +1,57 @@
+use std::pin::Pin;
+use std::sync::Mutex;
+
+use crate::task::{Context, Poll};
+
+/// Creates a stream that yields the given elements continually.
+///
+/// # Examples
+///
+/// ```
+/// # #![feature(async_await)]
+/// # fn main() { async_std::task::block_on(async {
+/// #
+/// use async_std::prelude::*;
+/// use async_std::stream;
+///
+/// let mut s = stream::cycle(vec![1, 2, 3]);
+///
```

Comment:
In std the API is:

```rust
let a = [1, 2, 3];

let mut it = a.iter().cycle();

assert_eq!(it.next(), Some(&1));
assert_eq!(it.next(), Some(&2));
assert_eq!(it.next(), Some(&3));
assert_eq!(it.next(), Some(&1));
assert_eq!(it.next(), Some(&2));
assert_eq!(it.next(), Some(&3));
assert_eq!(it.next(), Some(&1));
```

I'd expect this API to be (with #125 landed):
```rust
let a = [1, 2, 3];

let mut it = a.into_stream().cycle();

assert_eq!(it.next().await, Some(&1));
assert_eq!(it.next().await, Some(&2));
assert_eq!(it.next().await, Some(&3));
assert_eq!(it.next().await, Some(&1));
assert_eq!(it.next().await, Some(&2));
assert_eq!(it.next().await, Some(&3));
assert_eq!(it.next().await, Some(&1));
```

Similar, but subtly different in that it's more generic, which should make it more widely applicable!